providers/asrockrack/power.go

```diff
@@ -0,0 +1,104 @@
+package asrockrack
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	bmclibErrs "github.com/bmc-toolbox/bmclib/errors"
+	"github.com/pkg/errors"
+)
+
+type power struct {
+	Command int `json:"power_command"`
+}
+
+// PowerStateGet gets the power state of a machine
+func (a *ASRockRack) PowerStateGet(ctx context.Context) (state string, err error) {
+	info, err := a.chassisStatusInfo(ctx)
+	if err != nil {
+		return "", errors.Wrap(bmclibErrs.ErrPowerStatusRead, err.Error())
+	}
+
+	switch info.PowerStatus {
+	case 0:
+		return "Off", nil
+	case 1:
+		return "On", nil
+	default:
+		return "", errors.Wrap(
+			bmclibErrs.ErrPowerStatusRead,
+			fmt.Errorf("unknown status: %d", info.PowerStatus).Error(),
+		)
+	}
+}
+
+// PowerSet sets the hardware power state of a machine
+func (a *ASRockRack) PowerSet(ctx context.Context, state string) (ok bool, err error) {
+	switch strings.ToLower(state) {
+	case "on":
```

Comment:
Are these states documented or defined as constants somewhere? It would be nice to have a type alias for these

Reply:
yep, I've been thinking of this as well - created an issue to follow up https://github.com/bmc-toolbox/bmclib/issues/266